Library/ViewModels/DiscoveryPageViewModelTests.swift

```diff
@@ -1175,4 +1175,29 @@ internal final class DiscoveryPageViewModelTests: TestCase {
 
     XCTAssertEqual("ending_soon", segmentClientProps?["discover_sort"] as? String)
   }
+
+  func testBlockedUserNotification_RefreshesProjects() {
+    let playlist = (0...10).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let projectEnv = .template
+      |> DiscoveryEnvelope.lens.projects .~ playlist
+
+    withEnvironment(apiService: MockService(fetchDiscoveryResponse: projectEnv)) {
+      self.vm.inputs.configureWith(sort: .magic)
+      self.vm.inputs.viewWillAppear()
+      self.vm.inputs.viewDidAppear()
+      self.vm.inputs.selectedFilter(.defaults)
+
+      self.projectsAreLoading.assertValueCount(1)
+
+      self.scheduler.advance()
+
+      self.projectsAreLoading.assertValueCount(2)
+
+      self.vm.inputs.blockedUser()
+
+      self.scheduler.advance()
+
+      self.projectsAreLoading.assertValueCount(4)
```

Comment:
There's a similar test for `pullToRefresh` that requires this assert to be `       self.projectsAreLoading.assertValueCount(6)`. But that seems a bit odd to me (why the extra values)? I'm not sure, then, if this is correct or not.

Reply:
I think that might be because there's a loading indicator that shows for `pullToRefresh` (and I don't think we need to show it here). I could be wrong about that, since I didn't look thoroughly at that code. But test looks good to me!